x-pack/filebeat/input/entityanalytics/provider/azure/metrics.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package azure
+
+import "github.com/elastic/elastic-agent-libs/monitoring"
+
+type inputMetrics struct {
+	id       string
+	registry *monitoring.Registry
+
+	fullSyncTotal            *monitoring.Uint
+	fullSyncSuccess          *monitoring.Uint
+	fullSyncFailure          *monitoring.Uint
+	incrementalUpdateTotal   *monitoring.Uint
+	incrementalUpdateSuccess *monitoring.Uint
+	incrementalUpdateFailure *monitoring.Uint
+}
+
+func (m *inputMetrics) Close() {
+	m.registry.Remove(m.id)
+}
+
+func newMetrics(registry *monitoring.Registry, id string) *inputMetrics {
```

Comment:
Take a look how this is done in e.g. https://github.com/elastic/beats/blob/7356c98624eeac01b545804d0e10acfde8e180c7/x-pack/filebeat/input/lumberjack/metrics.go#L15-L45